File: src/main/webapp/WEB-INF/static/core/service/TimeService.js
/**
 * 
 */
app.factory('TimeService', ['$http', function($http) {
	
	var serverURL = function(url) {
		return SERVER_APP + '/time' + url;
	};	
	
	/**
	 * Pega time por id
	 */
	var _get = function(id) {
		return $http.get(serverURL('/get/'+id));
	};
	
	var _getTimes = function() {
		return $http.get(serverURL('/get/list'));
	};
	
	var _getListaPorEdicao = function(edicao) {
		return _getTimes();
	};
	
	/**
	 * Grava Time
	 */
	var _save = function(time) {
		return $http.post(serverURL('/save'), time);
	};
	
	/**
	 * Add jogador ao time
	 */
	var _addJogador = function(time, jogador) {
		return $http.post(serverURL("/"+time.id+'/add/jogador/'+jogador.id));
	};
	
	/**
	 * Remove jogador do time
	 */
	var _removeJogador = function(time, jogador) {
		return $http.post(serverURL("/"+time.id+'/remove/jogador/'+jogador.id));
	};
	
	return {
		
		get : _get,
		
		getTimes : _getTimes,
		
		getListaPorEdicao : _getListaPorEdicao,
		
		save : _save,
		
		addJogador : _addJogador,
		
		removeJogador : _removeJogador
		
	};
	
}]);

File: src/main/java/com/diegolirio/jcampeonato/service/TimeService.java
package com.diegolirio.jcampeonato.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.stereotype.Service;

import com.diegolirio.jcampeonato.dao.CampeonatoDao;
import com.diegolirio.jcampeonato.dao.TimeDao;
import com.diegolirio.jcampeonato.model.Jogador;
import com.diegolirio.jcampeonato.model.Time;

@Service("timeService")
public class TimeService extends AbstractGenericService<Time> {

	@Autowired @Qualifier("timeDao")
	private TimeDao timeDao;
	
	@Autowired @Qualifier("campeonatoDao")
	private CampeonatoDao campeonatoDao;

	@Autowired @Qualifier("jogadorService")
	private JogadorService jogadorService;

	/**
	 * Add jogador ao time
	 * @param time
	 * @param jogador
	 */
	public void addJogador(Time time, Jogador jogador) {
		List<Jogador> jogadores = this.jogadorService.getListByTime(time);
		jogadores.add(jogador);
		time.setJogadores(jogadores);
		this.timeDao.save(time);		
	}
	
	
	/**
	 * remove jogador do time
	 * @param time
	 * @param jogador
	 */
	public void removeJogador(Time time, Jogador jogador) {
		time.setJogadores(this.jogadorService.getListByTime(time));
		if(time.getJogadores().remove(jogador))
			this.timeDao.save(time);	
		else throw new RuntimeException("Não foi possivel remover Jogador");
	}

}
